### DISCUSSION TITLE
Sync session with BW CLI

### DISCUSSION THREAD

**Discussion by tyler-8** [ajrosen/Bitwarden-Accelerator#23] [General]
With the now deprecated https://github.com/blacs30/bitwarden-alfred-workflow there was a way to sync the Bitwarden unlock session used by the Alfred workflow with your BW CLI workflows via these shell commands: https://github.com/luckman212/bitwarden-cli-helpers/blob/master/functions.bash

I'm looking for a way to do the same with Bitwarden Accelerator. Is there a path to doing this today, or do the underlying workflows work too differently?

**Comment by ajrosen** [comment]
Interesting.  I see the functions don't really care if a session token is valid.  Does that matter?

Bitwarden Accelerator does not have a way to do this, but it wouldn't be too difficult to add.

**Reply by tyler-8** [reply]
My understanding is that it's just looking up the session token from the MacOS Keychain and assuming that it's correctly defined by the Alfred workflow so I don't think there'd need to be extra steps to validate the session token on the Bitwarden Accelerator side.

**Reply by tyler-8** [reply]
The overarching goal is to either

- Start a Bitwarden CLI session in the terminal, and then transfer/share that to Bitwarden Accelerator (instead of doing the unlock operation twice)
- Retrieve a Bitwarden CLI session from the keychain that was pre-defined by Bitwarden Accelerator in order to use the same session in the terminal

**Comment by ajrosen** [comment]
After a fair amount of testing, the answer is mostly no, this workflow can't support it.

The main problem is that _bw serve_, once started, does not use **BW_SESSION**.  It will read it when it starts, so the server will always start either **locked** or **unlocked**, depending on the session.

From then on the states are independent.  The server knows nothing about state changes made in the shell, and the shell can't see state changes made through the server.  So any saved session key is almost guaranteed to be wrong.

**Reply by tyler-8** [reply]
Ah yeah that's an interesting pickle. Well, I appreciate you looking at the possibility!
